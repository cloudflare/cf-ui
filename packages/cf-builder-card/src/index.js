import cardActions from './cardActions';
import CardActionTypes from './CardActionTypes';
import CardBuilder from './CardBuilder';
import cardReducer from './cardReducer';

export { cardActions, CardActionTypes, CardBuilder, cardReducer };
