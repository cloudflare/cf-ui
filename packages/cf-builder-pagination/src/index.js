import PaginationBuilder from './PaginationBuilder';

export { PaginationBuilder };
