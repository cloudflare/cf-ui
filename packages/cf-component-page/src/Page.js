import React, { PropTypes } from 'react';

class Page extends React.Component {
  render() {
    return (
      <article className="cf-page">
        {this.props.children}
      </article>
    );
  }
}

Page.propTypes = {
  children: PropTypes.node
};

export default Page;
