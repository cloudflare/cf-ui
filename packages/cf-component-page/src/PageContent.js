import React from 'react';
import PropTypes from 'prop-types';
import { createComponent } from 'cf-style-container';

const styles = () => ({
  marginLeft: 'auto',
  marginRight: 'auto',
  width: '90%',
  maxWidth: '79em'
});

const PageContent = ({ className, children }) => {
  return (
    <div className={className}>
      {children}
    </div>
  );
};

PageContent.propTypes = {
  className: PropTypes.string,
  children: PropTypes.node
};

export default createComponent(styles, PageContent);
