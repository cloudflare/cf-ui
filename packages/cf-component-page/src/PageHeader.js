const React = require('react');
const { PropTypes } = React;

class PageHeader extends React.Component {
  render() {
    return (
      <header className="cf-page__header">
        <h1 className="cf-page__title">{this.props.title}</h1>
        {this.props.subtitle &&
          <p className="cf-page__subtitle">{this.props.subtitle}</p>}
      </header>
    );
  }
}

PageHeader.propTypes = {
  title: PropTypes.string.isRequired,
  subtitle: PropTypes.string
};

module.exports = PageHeader;
