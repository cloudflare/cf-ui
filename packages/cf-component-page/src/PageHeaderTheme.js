import { checkBaseTheme } from 'cf-style-const';

export default baseTheme => {
  checkBaseTheme(baseTheme, 'PageHeaderTheme');
  return {
    marginBottom: 0,
    marginTop: 0,
    fontSize: '1.6em',
    defaultColor: baseTheme.defaultBackground
  };
};
