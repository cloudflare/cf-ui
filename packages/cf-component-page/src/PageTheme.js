export default () => ({
  paddingTop: 0,
  paddingBottom: 0
});
