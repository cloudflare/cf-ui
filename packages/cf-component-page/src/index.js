import Page from './Page';
import PageHeader from './PageHeader';

export { Page, PageHeader };
