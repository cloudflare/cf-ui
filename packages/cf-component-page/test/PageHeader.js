import React from 'react';
import renderer from 'react-test-renderer';
import { PageHeader } from '../src/index';
import felaSnapshot from 'cf-style-provider/src/felaSnapshot';

test('should render title', () => {
  const snapshot = felaSnapshot(<PageHeader title="Title" />);
  expect(snapshot.component).toMatchSnapshot();
  expect(snapshot.styles).toMatchSnapshot();
});

test('should render title/subtitle', () => {
  const snapshot = felaSnapshot(
    <PageHeader title="Title" subtitle="Subtitle" />
  );
  expect(snapshot.component).toMatchSnapshot();
  expect(snapshot.styles).toMatchSnapshot();
});
