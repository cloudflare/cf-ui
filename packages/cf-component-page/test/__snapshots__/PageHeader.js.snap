// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render title 1`] = `
<header
  className="cf-page__header"
>
  <h1
    className="cf-page__title"
  >
    Title
  </h1>
</header>
`;

exports[`should render title/subtitle 1`] = `
<header
  className="cf-page__header"
>
  <h1
    className="cf-page__title"
  >
    Title
  </h1>
  <h2
    className="cf-page__subtitle"
  >
    Subtitle
  </h2>
</header>
`;
